realm/realm-library/src/androidTest/java/io/realm/DynamicRealmTests.java

```diff
@@ -471,6 +507,129 @@ public void distinct_invalidTypes() {
     }
 
     @Test
+    public void distinctMultiArgs() {
+        final long numberOfBlocks = 25;
+        final long numberOfObjects = 10; // must be greater than 1
+        populateForDistinct(realm, numberOfBlocks, numberOfObjects, false);
+
+        //Simon Ask Ulsnes "I would expect it to be 25, because I would expect it to look for unique ​_combinations_​ or tuples."
+        RealmResults<DynamicRealmObject> distinctMulti = realm.distinct(AnnotationIndexTypes.CLASS_NAME, AnnotationIndexTypes.INDEX_FIELDS);
+        assertEquals(numberOfBlocks, distinctMulti.size());
+    }
+
+    @Test
+    public void distinctMultiArgs_switchedFieldsOrder() {
+        final long numberOfBlocks = 25; // must be greater than 1
```

Comment:
Perhaps add a comment that this expected to be the same because of the unique combinations behavior